source/tyr/tyr/resources.py

```diff
@@ -2053,66 +2053,111 @@ def delete(self, id=None):
+            args['page'], current_app.config.get('MAX_ITEMS_PER_PAGE', 5)
+        )
+
+        pagination_json = {
+            'current_page': pagination.page,
+            'items_per_page': pagination.per_page,
+            'total_items': pagination.total,
+        }
+
+        if pagination.has_next:
+            pagination_json['next'] = url_for(request.endpoint, page=pagination.next_num)
+        return marshal(pagination.items, streetnetwork_backend_fields), pagination_json
+
+    def get(self, backend_id=None):
+        resp = None
+        if backend_id:
+            resp = marshal(self._get_user_by_id(backend_id), streetnetwork_backend_fields)
         else:
-            return {'streetnetwork_backends': models.StreetNetworkBackend.all()}
+            resp = self._get_all_backends()
+            # In case of a response with pagination, the response type is a tuple and it needs to be serialized
+            if type(resp) == tuple:
+                return {'streetnetwork_backends': resp[0], 'pagination': resp[1]}
 
-    def put(self, id=None):
+        return {'streetnetwork_backends': resp}
+
+    def _validate_input(self, json_data):
         """
-        Create or update a streetnetwork backend in db
+        Check that the data received contains all required info
+        :param json_data: data received in request
         """
+        try:
+            validate(json_data, streetnetwork_backend_format)
+        except ValidationError as e:
+            abort(400, status="invalid data", message='{}'.format(parse_error(e)))
 
-        def _validate_input(json_data):
-            """
-            Check that the data received contains all required info
-            :param json_data: data received in request
-            """
-            try:
-                validate(json_data, streetnetwork_backend_format)
-            except ValidationError as e:
-                abort(400, status="invalid data", message='{}'.format(parse_error(e)))
+    def post(self, backend_id=None):
+        """
+        Create a streetnetwork backend in db
+        """
+        if not backend_id:
+            abort(400, status="error", message='backend_id is required')
 
-        if not id:
-            abort(400, status="error", message='id is required')
+        try:
+            input_json = request.get_json(force=True, silent=False)
+        except BadRequest:
+            abort(400, status="error", message='Incorrect json provided')
+
+        self._validate_input(input_json)
+        sn_backend = models.StreetNetworkBackend(backend_id, input_json)
+        try:
+            models.db.session.add(sn_backend)
+            models.db.session.commit()
+        except sqlalchemy.exc.IntegrityError as ex:
+            abort(400, status="error", message=str(ex))
+
+        resp = marshal(sn_backend, streetnetwork_backend_fields), 201
+        return {'streetnetwork_backend': resp[0]}, resp[1] if resp[1] == 201 else resp
```

Comment:
since the resp[1] is always 201... it's weird that you still need to check it here...